### ISSUE TITLE
BUG after updating from 30/1 to 2/2 - fs unmountable

### ISSUE THREAD

**Issue by jpsollie** [koverstreet/bcachefs#188]
I updated bcachefs code from 29/1 to 2/2, and this made the filesystem unmountable.
- Compiled & installed bcachefs module
- unmounted all bcachefs filesystems
- unload module & reload module
- mount filesystem
And here it hangs with a kernel bug:
```
src@linuxserver/linuxserver kernel: sd 18:0:1:0: [sdf] Very big device. Trying to use READ CAPACITY(16).
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: bcachefs (6487fff2-c26e-4d19-916f-8dba75737547): recovering from clean shutdown, journal seq 150161
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: ------------[ cut here ]------------
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: Kernel BUG at bch2_journal_write+0x9c9/0xbd0 [bcachefs] [verbose debug info unavailable]
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: invalid opcode: 0000 [#1] SMP
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: CPU: 27 PID: 790 Comm: kworker/27:1H Tainted: G S                5.10.12+ #1
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: Hardware name: Gigabyte Technology Co., Ltd. X399 DESIGNARE EX/X399 DESIGNARE EX-CF, BIOS F12 12/11/2019
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: Workqueue: events_highpri bch2_journal_write [bcachefs]
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RIP: 0010:bch2_journal_write+0x9c9/0xbd0 [bcachefs]
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: Code: 0f fe ff ff 49 8b 46 78 f0 48 83 28 01 0f 85 c2 fe ff ff 49 8b 46 78 49 8d 7e 70 ff 50 08 e9 b2 fe ff ff e9 9f 1d 02 00 0f 0b <0f> 0b 0f 0b 48 ff cb be ff ff ff ff 4c 89 f7 48 c1 eb 0c 48 0f bd
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RSP: 0018:ffff888112bd3d50 EFLAGS: 00010297
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RAX: 00000000000000dc RBX: ffff88811d370000 RCX: 0000000000000002
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RDX: 0000000000000000 RSI: 0000000000000286 RDI: ffff8881178a3198
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RBP: ffff888112bd3e78 R08: ffff8881178a3190 R09: ffff888112bd3ca0
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: R10: fffffffffffffff8 R11: 0000000000000018 R12: ffff8881178b4e20
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: R13: ffff88811d370038 R14: ffff88811d370040 R15: ffff8881178a0000
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: FS:  0000000000000000(0000) GS:ffff88885eec0000(0000) knlGS:0000000000000000
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: CR2: 00007fc8b03d660c CR3: 00000001043e9000 CR4: 00000000003506e0
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: Call Trace:
Feb  3 08:00:23 src@linuxserver/linuxserver kernel:  ? elv_rb_del+0x25/0x40
Feb  3 08:00:23 src@linuxserver/linuxserver kernel:  ? sched_clock_cpu+0x17/0xc0
Feb  3 08:00:23 src@linuxserver/linuxserver kernel:  ? newidle_balance.isra.0+0x187/0x300
Feb  3 08:00:23 src@linuxserver/linuxserver kernel:  ? process_one_work+0x1a6/0x300
Feb  3 08:00:23 src@linuxserver/linuxserver kernel:  ? worker_thread+0x45/0x3c0
Feb  3 08:00:23 src@linuxserver/linuxserver kernel:  ? current_work+0x30/0x30
Feb  3 08:00:23 src@linuxserver/linuxserver kernel:  ? kthread+0xf2/0x130
Feb  3 08:00:23 src@linuxserver/linuxserver kernel:  ? kthread_park+0x80/0x80
Feb  3 08:00:23 src@linuxserver/linuxserver kernel:  ? ret_from_fork+0x1f/0x30
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: Modules linked in: bcachefs i2c_dev btrfs blake2b_generic xor raid6_pq ses input_leds uas amd_energy it87 hwmon_vid iwlmvm kvm_amd mac80211 kvm irqbypass ghash_clmulni_intel iwlwifi aesni_intel usb_storage glue_helper cfg80211 igb aacraid rapl btusb amdgpu wmi_bmof btintel hid_generic gpu_sched ttm sp5100_tco i2c_piix4 tpm_crb tpm_tis pinctrl_amd tpm_tis_core efivarfs [last unloaded: bcachefs]
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: ---[ end trace da05a674fa0f0108 ]---
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RIP: 0010:bch2_journal_write+0x9c9/0xbd0 [bcachefs]
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: Code: 0f fe ff ff 49 8b 46 78 f0 48 83 28 01 0f 85 c2 fe ff ff 49 8b 46 78 49 8d 7e 70 ff 50 08 e9 b2 fe ff ff e9 9f 1d 02 00 0f 0b <0f> 0b 0f 0b 48 ff cb be ff ff ff ff 4c 89 f7 48 c1 eb 0c 48 0f bd
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RSP: 0018:ffff888112bd3d50 EFLAGS: 00010297
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RAX: 00000000000000dc RBX: ffff88811d370000 RCX: 0000000000000002
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RDX: 0000000000000000 RSI: 0000000000000286 RDI: ffff8881178a3198
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: RBP: ffff888112bd3e78 R08: ffff8881178a3190 R09: ffff888112bd3ca0
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: R10: fffffffffffffff8 R11: 0000000000000018 R12: ffff8881178b4e20
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: R13: ffff88811d370038 R14: ffff88811d370040 R15: ffff8881178a0000
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: FS:  0000000000000000(0000) GS:ffff88885eec0000(0000) knlGS:0000000000000000
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
Feb  3 08:00:23 src@linuxserver/linuxserver kernel: CR2: 00007fc8b03d660c CR3: 00000001043e9000 CR4: 00000000003506e0
Feb  3 08:01:46 src@linuxserver/linuxserver kernel: BUG: workqueue lockup - pool cpus=27 node=0 flags=0x0 nice=-20 stuck for 47s!
Feb  3 08:01:46 src@linuxserver/linuxserver kernel: Showing busy workqueues and worker pools:
Feb  3 08:01:46 src@linuxserver/linuxserver kernel: workqueue events: flags=0x0
Feb  3 08:01:46 src@linuxserver/linuxserver kernel:   pwq 0: cpus=0 node=0 flags=0x0 nice=0 active=2/256 refcnt=3
Feb  3 08:01:46 src@linuxserver/linuxserver kernel:     pending: vmstat_shepherd, drm_fb_helper_dirty_work
Feb  3 08:01:46 src@linuxserver/linuxserver kernel: workqueue events_highpri: flags=0x10
Feb  3 08:01:46 src@linuxserver/linuxserver kernel:   pwq 55: cpus=27 node=0 flags=0x0 nice=-20 active=1/256 refcnt=2
Feb  3 08:01:46 src@linuxserver/linuxserver kernel:     in-flight: 790:bch2_journal_write [bcachefs]
Feb  3 08:01:46 src@linuxserver/linuxserver kernel: workqueue kblockd: flags=0x18
Feb  3 08:01:46 src@linuxserver/linuxserver kernel:   pwq 55: cpus=27 node=0 flags=0x0 nice=-20 active=4/256 refcnt=5
Feb  3 08:01:46 src@linuxserver/linuxserver kernel:     pending: blk_mq_timeout_work, blk_mq_timeout_work, blk_mq_timeout_work, blk_mq_timeout_work
Feb  3 08:01:46 src@linuxserver/linuxserver kernel: pool 55: cpus=27 node=0 flags=0x0 nice=-20 hung=47s workers=2 idle: 177
```
Even after reboot, the filesystem is unmountable with current bcachefs module.  Downgrading to version 29/1 fixes this.
I'll try to bisect it asap, but not sure which commits could be bisected on its own

**Comment by jpsollie** [comment]
reverting to commit 84d658fdc128075035df9b018d0f62ebc1ad2b15 fixed the issue

**Comment by koverstreet** [comment]
Fixed by 79847e4824
